### ISSUE TITLE
CRITICAL - Webhook Malfunctioning for our client

### ISSUE THREAD

**Issue by rajeshwl** [reapit/foundations#7304]
We have requested an issue in webhook with other client tempocasa(TMC) yesterday, and that has been fixed yesterday. But now we started receiving requests in other endpoints configured to different clients. We are receiving data of Lomond Group(LOC) in other client endpoints.

Clients now receiving wrong data are:
Aberdein Considine - AAC
Bentleyhurst - BEH

Reference of the old issue pulse: https://github.com/reapit/foundations/issues/7288

**Comment by rajeshwl** [comment]
One of the tenancy request received is given below.
```
{
   "eventId":"1d65f12d-db91-4528-b94d-96235eb98971",
   "entityId":"ALM220332",
   "customerId":"LOC-GTHG",
   "eventTime":"2022-07-15T07:43:33.564733Z",
   "topicId":"tenancies.modified",
   "new":{
      "id":"ALM220332",
      "created":"2022-07-15T07:43:03Z",
      "modified":"2022-07-15T07:43:29Z",
      "startDate":"2022-09-01",
      "endDate":"2023-08-31",
      "status":"offerPending",
      "agentRole":"managed",
      "rent":1500,
      "rentFrequency":"monthly",
      "endDateConfirmed":false,
      "isPeriodic":false,
      "rentInstalmentsFrequency":"monthly",
      "rentInstalmentsAmount":1500,
      "rentInstalmentsStart":null,
      "meterReadingGas":null,
      "meterReadingGasLastRead":null,
      "meterReadingElectricity":null,
      "meterReadingElectricityLastRead":null,
      "meterReadingWater":null,
      "meterReadingWaterLastRead":null,
      "typeId":"AS",
      "negotiatorId":"SEP",
      "propertyId":"ZXA120052",
      "applicantId":"ALM214861",
      "lettingFee":{
         "type":"fixed",
         "amount":650,
         "frequency":"upfront"
      },
      "managementFee":{
         "type":"percentage",
         "amount":5,
         "frequency":"sameAsLettingFee"
      },
      "source":{
         "id":"SRC055",
         "type":"source"
      },
      "related":[
      ],
      "fromArchive":false,
      "_eTag":"\"46E6B2AE3CE7B7CB94512F08DAE7D004\""
   },
   "old":{
      "typeId":"AS",
      "endDate":null,
      "rentInstalmentsAmount":0,
      "status":"offerPending",
      "meterReadingGas":null,
      "rentInstalmentsStart":null,
      "applicantId":"ALM214861",
      "rentInstalmentsFrequency":"monthly",
      "lettingFee":{
         "type":"fixed",
         "amount":650,
         "frequency":"upfront"
      },
      "startDate":null,
      "rentFrequency":"monthly",
      "id":"ALM220332",
      "agentRole":"managed",
      "modified":"2022-07-15T07:43:03Z",
      "meterReadingWater":null,
      "isPeriodic":false,
      "meterReadingGasLastRead":null,
      "propertyId":"ZXA120052",
      "meterReadingElectricity":null,
      "fromArchive":false,
      "created":"2022-07-15T07:43:03Z",
      "endDateConfirmed":false,
      "source":{
         "type":"source",
         "id":"SRC055"
      },
      "related":[
      ],
      "meterReadingElectricityLastRead":null,
      "_eTag":"\"3A90A7A25ADC3E6663452889B1FB9C65\"",
      "meterReadingWaterLastRead":null,
      "managementFee":{
         "type":"percentage",
         "amount":5,
         "frequency":"sameAsLettingFee"
      },
      "negotiatorId":"SEP",
      "rent":1500
   },
   "diff":{
      "endDate":[
         null,
         "2023-08-31"
      ],
      "rentInstalmentsAmount":[
         0,
         1500
      ],
      "startDate":[
         null,
         "2022-09-01"
      ],
      "modified":[
         "2022-07-15T07:43:03Z",
         "2022-07-15T07:43:29Z"
      ],
      "_eTag":[
         "\"3A90A7A25ADC3E6663452889B1FB9C65\"",
         "\"46E6B2AE3CE7B7CB94512F08DAE7D004\""
      ]
   },
   "SendAttempts":1
}
```

**Comment by plittlewood-rpt** [comment]
Hi @rajeshwl which endpoint did you receive this to?

**Comment by rajeshwl** [comment]
@plittlewood-rpt ,
Currently we have started receiving data in the below endpoints.
https://bentleyhurst2-strapi.q.starberry.com/reapits
https://acandco-strapi.q.starberry.com/reapits

Currently we have enabled only 3 endpoints. So kindly make sure this does not repeat to the new endpoints we create in future  as well.

**Comment by plittlewood-rpt** [comment]
Hi @rajeshwl  - I can see a subscription to LOC for both these endpoints. Can you not see these in the UI?

**Comment by plittlewood-rpt** [comment]
OK thanks - I will clean these up in the database now and we will need to understand how they got there in the first place (and ultimately why you can't see them). Do you always use the developer portal to setup your webhooks or do you make use of our resthooks endpoints?

**Comment by rajeshwl** [comment]
We always use developer portal to setup webhooks. We are not using resthooks as of now.

**Comment by plittlewood-rpt** [comment]
Thanks for confirming. I've deleted these subscriptions - it looks like it's something to do with the office group mechanism that's preventing these from being displayed in the UI. We'll run some tests here to get to the bottom of it. These messages should now stop

**Comment by plittlewood-rpt** [comment]
Related to https://github.com/reapit/foundations/issues/7253

**Comment by plittlewood-rpt** [comment]
@rajeshwl I can see this behaviour stopped as expected. I will close this ticket as the underlying issue will be resolved in #7253.
